tests/integration/optlyplugins/notification_manager.go

```diff
@@ -17,48 +17,43 @@
 package optlyplugins
 
 import (
+	"github.com/optimizely/go-sdk/pkg/client"
 	"github.com/optimizely/go-sdk/pkg/decision"
 	"github.com/optimizely/go-sdk/pkg/entities"
+	"github.com/optimizely/go-sdk/pkg/event"
 	"github.com/optimizely/go-sdk/pkg/notification"
 	"github.com/optimizely/go-sdk/tests/integration/models"
 )
 
-// TestCompositeService represents a CompositeService with custom implementations
-type TestCompositeService struct {
-	decision.CompositeService
-	listenersCalled []models.DecisionListener
+// NotificationManager manager class for notification listeners
+type NotificationManager struct {
+	DecisionService *decision.CompositeService
+	Client          *client.OptimizelyClient
```

Comment:
Instead of retaining an instance of the client, maybe we can just pass it into `SubscribeNotifications`. I just think it's weird that the client wrapper has a reference to the client and this also has a reference to the client.